lexicon/__main__.py

```diff
@@ -36,13 +38,10 @@ def BaseProviderParser():
     return parser
 
 def MainParser():
-    current_filepath = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'providers')
-    providers = [os.path.splitext(f)[0] for f in os.listdir(current_filepath) if os.path.isfile(os.path.join(current_filepath, f))]
-    providers = list(set(providers))
-    providers.remove('base')
-    providers.remove('__init__')
-    providers = [x for x in providers if not x.startswith('.')]
-
+    providers = []
+    for _, modname, _ in pkgutil.iter_modules(providers_package.__path__):
+        if modname != 'base':
+            providers.append(modname)
```

Comment:
do we have to filter out the `__init__` file like we do in the old code?

Reply:
In fact your question is closely related to the other one about absolute imports and the fact that absolute imports are not tied to an actual filesystem architecture. `pkgutil.iter_modules` does not care about the Python files, it cares about what is resolved as packages and modules in the current Python runtime.

Namely, it is: 'iterate over all the modules available in the given package'. In that extend, `__init__` does not exists. It is just a file containing the logic to construct the relevant package, but it is not itself a module. So it does not appear in the iteration.

What are the other advantages ? Well, imagine that something (certainly `__init__.py`) construct and dynamically add a module in a package. If you rely on the filesystem, you will not see the dynamic module, because there is not file for it. But if you call `pkgutil.iter_modules`, you will see it.